### DISCUSSION TITLE
Setup

### DISCUSSION THREAD

**Discussion by euriska** [nautechsystems/nautilus_trader#1112] [Q&A] [answered]
Where can I find more detailed instructions on how to setup the various aspects of Nautilus such as the back testing and trading. A suggested architecture layout would be helpful.

Thanks,

**Comment by limx0** [comment]
Hi @euriska - I'm not exactly sure what you're asking - but there is an examples folder in the repo with a handful of backtesting and live trading example setups. The readme also contains a high level architectural diagram.

**Comment by euriska** [comment]
The architectural diagram is nice but it doesn't tell me what servers I need to build or if it's one server. I would think the back testing server and execution engines would be on separate servers at the very least. I did not see any guidance on how to build these components.

Thanks,
________________________________

**Reply by limx0** [reply]
Nautilus is run as a single process. Typically backtest and live execution would however be split across different servers. 

Nautilus is built as a single python package containing live and backtest functionality

**Comment by euriska** [comment]
And is there guidance on how to build these additional components and integrate them into the messaging bus and cache. I would prefer to build the environment in its final end state.

Thanks,
________________________________

**Reply by limx0** [reply, marked as answer]
The "components" are all internal to the nautilus process - no building required. Running any of the live or backtests examples includes all of these components.

**Comment by euriska** [comment]
Brad,
 How to I accomplish this: "Typically backtest and live execution would however be split across different servers."
________________________________
